homeassistant/components/netatmo/climate.py

```diff
@@ -82,90 +151,264 @@ def temperature_unit(self):
     @property
     def current_temperature(self):
         """Return the current temperature."""
-        return self._data.current_temperature
+        return self._data.room_status[self._room_id]['current_temperature']
 
     @property
     def target_temperature(self):
         """Return the temperature we try to reach."""
-        return self._target_temperature
+        return self._data.room_status[self._room_id]['target_temperature']
 
     @property
     def current_operation(self):
         """Return the current state of the thermostat."""
-        state = self._data.thermostatdata.relay_cmd
-        if state == 0:
+        state = self._data.room_status[self._room_id]['heating_status']
+        if state is False:
             return STATE_IDLE
-        if state == 100:
+        if state is True:
             return STATE_HEAT
+        return STATE_UNKNOWN
```

Comment:
`current_operation` should return `self._operation_mode`. That's our current architecture for climate component. We have to follow that.

Reply:
Thanks for your comment. I tried this and tested on my HA, and found out that in the UI it cannot show the status if it is heating or not. The current version can show the operation mode with a highlighted icon on the bottom and a correct heating status with words. And this is not introduced by me, the previous code also does not return its operation_mode, but the boiler status. I think we should think about adding another property to describe the working mode/status, which may be different from operation mode, indicating the real working status of the climate device.

Reply:
Yes, this is a limitation of our current architecture. See https://github.com/home-assistant/architecture/issues/22 for the architecture issue where we discuss the future of the climate component.

For now we should change this property to return according to our current architecture. Please open a new PR where we address this, which will be a breaking change.

Reply:
OK. Modified accordingly and created PR 21633.